src/codehelp/codehelp.vala

```diff
@@ -20,6 +20,272 @@
+        }
+        // SEARCH_PATH_FROM_ENVP does not seem to be available even in quite fast distros like Fedora 35
+        var launcher = new SubprocessLauncher (SubprocessFlags.STDERR_PIPE | SubprocessFlags.STDOUT_PIPE | SubprocessFlags.STDIN_PIPE);
+        var env_vars = Environ.get ();
+        for (var i = 0; i < env_vars.length; i++) {
+            var env = env_vars[i];
+            // Patch the PATH variable to include some other "standard paths"
+            if (env.has_prefix ("PATH=") && env.contains ("/usr/bin")) {
+                env_vars[i] = env + ":/usr/local/bin";
+            }
+        }
+        launcher.set_environ (env_vars);
+        Subprocess subprocess;
+        try {
+            subprocess = launcher.spawn ("uncrustify", "-c", config.get_path (), "-l", "vala", null);
+            uint8[] contents;
+            FileUtils.get_data (source.filename, out contents);
+            size_t bytes_written;
+            subprocess.get_stdin_pipe ().write_all (contents, out bytes_written);
+            subprocess.get_stdin_pipe ().close ();
+        } catch (Error e) {
+            throw new FormattingError.SPAWN_ERROR ("Spawning: " + e.message);
+        }
+        var str = "";
+        var n = 0;
+        try {
+            var dis = new DataInputStream (subprocess.get_stdout_pipe ());
+            var sb = new StringBuilder ();
+            string tmp = null;
+            size_t len;
+            while ((tmp = dis.read_line (out len)) != null) {
+                sb.append (tmp).append_c ('\n');
+                n++;
+            }
+            str = sb.str;
+            subprocess.wait ();
+        } catch (Error e) {
+            throw new FormattingError.FORMATTING_ERROR ("Formatting: " + e.message);
+        }
+        var status = subprocess.get_exit_status ();
+        if (status != 0) {
+            try {
+                var dis = new DataInputStream (subprocess.get_stderr_pipe ());
+                string tmp = null;
+                size_t len;
+                while ((tmp = dis.read_line (out len)) != null) {
+                    warning ("[Uncrustify stderr]: %s", tmp);
+                }
+            } catch (Error e) {
+                throw new FormattingError.FORMATTING_ERROR ("Error while gathering error data as uncrustify failed: %s", e.message);
+            }
+            throw new FormattingError.FORMATTING_ERROR ("uncrustify failed with code %d", status);
+        }
+        return new Lsp.TextEdit () {
+            range = new Lsp.Range () {
+                start = new Lsp.Position () {
+                    line = 0,
+                    character = 0
+                },
+                end = new Lsp.Position () {
+                    line = n + 1,
+                    // Just for the trailing newline
+                    character = 1
+                }
+            },
+            newText = str
+        };
+    }
+
+    File get_uncrustify_config (Lsp.FormattingOptions options) throws Error {
```

Comment:
Are you sure you can't return a `string[]` argument list here and then use `launcher.spawnv()` instead? Then you could call this function `build_uncrustify_args()`.

Reply:
I'm quite sure (75%) after browsing the code and playing around, that I can't pass the equivalent of `set func_call_user = C_ NC_ N_ Q_ _` just via CLI-Arguments. We could simply drop this line, but then e.g. `_("foo")` would be simply formatted as `_ ("foo")`

Reply:
I think it's fine. `_ (` is something that appears in real code.